src/libraries/System.Text.RegularExpressions/tests/PrecompiledRegexScenarioTest.cs

```diff
@@ -18,12 +18,37 @@ public class PrecompiledRegexScenarioTest
         public void TestPrecompiledRegex()
         {
             string text = "asdf134success1245something";
+            string textWithMultipleMatches = @"asdf134success1245something
+bsdf135success1245somethingelse
+csdf136success2245somethingnew
+dsdf137success3245somethingold";
+            string[] expectedMatches = textWithMultipleMatches.Split(Environment.NewLine);
             RegexTestClass testClass = new RegexTestClass();
 
+            // Test Matches overloads
             Assert.Equal(1, testClass.Matches(text).Count);
+            Assert.Equal(0, testClass.Matches(text, startat: 7).Count);
+            MatchCollection multipleMatches = testClass.Matches(textWithMultipleMatches);
+            Assert.Equal(4, multipleMatches.Count);
+            for (int i = 0; i < expectedMatches.Length; i++)
+            {
+                Assert.Equal(expectedMatches[i], multipleMatches[i].Value.Trim()); // Calling Trim since the match will contain the new line as part of the match.
+            }
+
+
+            // Test Match overloads
             Assert.Equal(1, testClass.Match(text).Groups[0].Captures.Count);
+            Assert.Equal(Match.Empty, testClass.Match(text, beginning: 7, length: text.Length - 7));
+            Assert.Equal(5, testClass.Match(text, beginning: 5, length: text.Length - 5).Index);
+            Assert.Equal(Match.Empty, testClass.Match(text, startat: 7));
+            Assert.Equal(6, testClass.Match(text, startat: 6).Index);
+
+            // Test Span-based IsMatch throws for Precompiled scenario.
+            Assert.Throws<NotSupportedException>(() => testClass.IsMatch(text.AsSpan()));
+
+            // Test Groups
             Assert.Equal(text, testClass.Match(text).Groups[0].Value);
-            Assert.Equal(new int[] { 0, 1, 2}, testClass.GetGroupNumbers());
+            Assert.Equal(new int[] { 0, 1, 2 }, testClass.GetGroupNumbers());
             Assert.Equal(new string[] { "0", "1", "output" }, testClass.GetGroupNames());
```

Comment:
Can you also add use of Replace, Split, and Count, since those go through different code paths from IsMatch/Match/Matches?